src/Task/PhpCsFixerV2.php

```diff
@@ -91,6 +86,8 @@ public function run(ContextInterface $context)
             return $this->runOnAllFiles($context, $arguments);
         }
 
+        $arguments->add('--path-mode=intersection');
```

Comment:
> As you mentioned, currently the best option is to handle errors with intersection and an empty finder.

and that means you need to load config from file and change it, it finder is not fully configured.
but that, again, would require grumphp to understand the code it's executing (know the classes etc), which, as other solution I suggested, you refused.
you want to be aware of configuration but refuse to know it classes. not much you can do with that restriction (except run cli up to 2 times approach)